Remotion/Web/ClientScript/Scripts/SmartPage.ts

```diff
@@ -258,17 +263,55 @@ class SmartPage_Context
     }
   };
 
-  public SmartPage_PageRequestManager_endRequest = function (sender: unknown, args: Sys.WebForms.EndRequestEventArgs)
+  public SmartPage_PageRequestManager_endRequest = (sender: unknown, args: Sys.WebForms.EndRequestEventArgs) =>
   {
-    if (args.get_error() != undefined && (args.get_error() as any).httpStatusCode == '500')
+    if (args.get_error())
     {
-      const errorMessage = args.get_error().message;
-      args.set_errorHandled(true);
+      let statusCode = (args.get_error() as any).httpStatusCode;
+
+      if (!statusCode || statusCode === 200)
+      {
+        this.HandleRequestError("undefined", args);
+      }
+      else
+      {
+        this.HandleRequestError(statusCode, args);
+      }
+    }
+  }
+
+  private HandleRequestError(statusCode: string, args: Sys.WebForms.EndRequestEventArgs): void
+  {
+    let response = args.get_response();
+    let errorMessage = args.get_error().message;
 
-      const errorBody = '<div class="SmartPageErrorBody"><div>' + errorMessage + '</div></div>';
+    //@ts-ignore
+    let serverError: string = Sys.WebForms.Res.PRM_ServerError;
+    let isSynchronousError = errorMessage.includes(serverError.substring(0, serverError.indexOf(':')));
 
-      SmartPage_Context.Instance!.ShowMessage("SmartPageServerErrorMessage", errorBody);
+    let errorBody = '<div class="SmartPageErrorBody"><div>';
+
+    if (isSynchronousError)
+    {
+      errorBody += errorMessage + ', ' + response.get_statusText() + '<br>';
+    }
+    else
+    {
+      const shortenedErrorMessage = errorMessage.substring(errorMessage.indexOf(':') + 1);
+      errorBody += shortenedErrorMessage;
     }
+
+    if (isSynchronousError)
+    {
+      errorBody += '<iframe sandbox style="width: 100%; height: 80%; border: hidden;" srcdoc="<div>';
```

Comment:
when there's a synchrous type error, the content is a complete html page. there should be no DIV wrapping the content.